### ISSUE TITLE
Dashboard Issues

### ISSUE THREAD

**Issue by eseglem** [Zac-HD/hypofuzz#31]
When looking at a specific test, the "Minimal covering examples" just shows the same example repeated dozens of times. I know it is going through a bunch of different ones from the cache, and not finding any more coverage. Just not displaying all of them. 

And  "See patches with covering and/or failing examples" throws a `ValueError` complaining about the data frame not having any columns. 

I am running it on Python 3.11, and these are the versions of packages I think could be relevant. They are the latest version of everything except `pytest`, which has a compatibility issue. 

```
pytest == 7.4.4
hypofuzz == 23.12.1
hypothesis == 6.93.2
dash == 2.15.0
flask == 3.0.1
jinja2 == 3.1.3
numpy == 1.26.3
pandas == 2.0.3
plotly == 5.18.0
```

**Comment by Zac-HD** [comment]
The repeated minimal-covering-examples was because they're different at the level of Hypothesis' internal representation... but apparently not by their `repr`s.  I've added a dedupe-by-string-form step in the just-released version 24.02.1.

I'm not sure why the see-patches link would be failing; there's explicitly handling for the case where we don't have any patches to display.  Is this right at startup, or after you've left it running for a few minutes?

**Comment by eseglem** [comment]
Ok, updated to latest on hypothesis (6.97.4) and hypofuzz(24.2.1). 

I no longer see the same example repeated dozens of times, but I only see one now. Which is confusing me. I used to see something like 160+ examples. Lots of small ones which cover different branches. It is replaying saved examples at the beginning going through them and getting new branches on each, so I would expect to see all of them on the examples page. For a while there it was displaying a single mega example that covered a large number of branches. And I thought it might be trying to cover as many as possible in one, but eventually that example changed to be one that definitely does not cover the same branches. 

For additional context: I am fuzzing a Lark grammar. Based on the lack of various keywords, I can tell that some of the examples are not able to cover every branch. I did have to write a custom Strategy for it [here](https://github.com/eseglem/pycql2/blob/develop/tests/test_pycql2_hypothesis.py), but I do not think anything I did in there is related. 

I also re-named the `.hypothesis` folder so it would be starting from scratch, in case the cache got messed up somehow. And still seeing that same behavior. It is, of course, finding lots of new branches, but each time the minimal examples only shows what I assume is the most recent one. Not really sure the right way to hook into everything and actually verify that though.

I dug a little more and that appears to be an issue with clicking the link rather than the page itself. It is pointing to [dashboard.py#L127](https://github.com/Zac-HD/hypofuzz/blob/master/src/hypofuzz/dashboard.py#L127) and throwing `ValueError: Value of 'x' is not the name of a column in 'data_frame'. Expected one of [] but received: ninputs` so it appears that `trace` is empty at that point. And it is not getting to the patch page itself. If I open a new tab and directly go to http://localhost:9999/patches/ that page loads.

**Comment by Zac-HD** [comment]
Can you show me the before and after of your minimal-covering-examples page?  That strategy looks good (very nice in fact!), but I suspect I'm missing something here.  The recent change was intended to avoid printing identical-appearing examples generated by different underlying choice sequences, but should have left visually-distinct entries alone.

I think the solution to the `ValueError` is just going to be returning a short "nothing on this page yet, check back soon" page if `trace` is empty for that test.

**Comment by Zac-HD** [comment]
Thanks so much!  Shipped a fix in d2c548c44bcdcd15cffbc1ff2f4e40c6fd9f2d85 😁
